public/hooks/use_feed_back.tsx

```diff
@@ -3,48 +3,48 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { useState } from 'react';
-import { ASSISTANT_API } from '../../common/constants/llm';
+import { useState, useEffect } from 'react';
+
 import { IOutput, Interaction } from '../../common/types/chat_saved_object_attributes';
-import { useCore } from '../contexts/core_context';
 import { useChatState } from './use_chat_state';
-import { SendFeedbackBody } from '../../common/types/chat_saved_object_attributes';
+import { getIncontextInsightRegistry } from '../services';
 
-export const useFeedback = (interaction?: Interaction | null) => {
-  const core = useCore();
-  const { chatState } = useChatState();
+export const useFeedback = (interaction?: Interaction | null, hasChatState: boolean = true) => {
+  const registry = getIncontextInsightRegistry();
+  const chatStateContext = useChatState();
   const [feedbackResult, setFeedbackResult] = useState<undefined | boolean>(
     interaction?.additional_info?.feedback?.satisfaction ?? undefined
   );
 
   const sendFeedback = async (message: IOutput, correct: boolean) => {
-    const outputMessage = message;
-    // Markdown type output all has interactionId. The interactionId of message is equal to interaction id.
-    const outputMessageIndex = chatState.messages.findIndex((item) => {
-      return item.type === 'output' && item.interactionId === message.interactionId;
-    });
-    const inputMessage = chatState.messages
-      .slice(0, outputMessageIndex)
-      .reverse()
-      .find((item) => item.type === 'input');
-    if (!inputMessage) {
-      return;
+    if (hasChatState && chatStateContext?.chatState) {
+      const chatState = chatStateContext.chatState;
+      // Markdown type output all has interactionId. The interactionId of message is equal to interaction id.
+      const outputMessageIndex = chatState.messages.findIndex((item) => {
+        return item.type === 'output' && item.interactionId === message.interactionId;
+      });
+      const inputMessage = chatState.messages
+        .slice(0, outputMessageIndex)
+        .reverse()
+        .find((item) => item.type === 'input');
+      if (!inputMessage) {
+        return;
+      }
     }
+    registry.sendFeedbackRequest(message.interactionId, correct);
```

Comment:
> I strongly feel we can add a new assistant_service to aggregate POST/GET APIs in services folder. So assistant_service wraps those APIs and whoever call them can reuse its operations.

But for now, we can keep the implementation of directly calling POST/PUT and seek additional suggestions from other reviewer as this might introduce overhead of refactoring existing codes.